src/hdmf/build/warnings.py

```diff
@@ -15,6 +15,13 @@ class IncorrectQuantityBuildWarning(BuildWarning):
     pass
 
 
+class IncorrectDatasetShapeBuildWarning(BuildWarning):
```

Comment:
All of these warnings should really be errors because they result in an invalid file. @oruebel what do you think? Should we upgrade all of these warnings to errors?

Reply:
All except for `DtypeConversionWarning` and the generic `BuildWarning`

Reply:
And maybe also the `OrphanContainerWarning`.